server/services/token/token_service.go

```diff
@@ -49,26 +38,26 @@ func DecodeToken(tokenString string) (Claims, error) {
 	if err != nil || !parsedToken.Valid {
 		if claims.ExpiresAt != nil && (*claims.ExpiresAt).Before(time.Now()) {
 			var userToBeLoggedOut model.User
-			err := db.FindById(&userToBeLoggedOut, claims.UserId, "user_id")
+			err := db.FindById(&userToBeLoggedOut, claims.Id, "user_id")
 			if err != nil {
-				return *claims, fmt.Errorf("error finding user in db")
+				return nil, fmt.Errorf("error finding user in db")
 			}
-			query := "UPDATE users SET is_active = false WHERE user_id = '" + claims.UserId + "'"
+			query := "UPDATE users SET is_active = false WHERE user_id = '" + claims.Id + "'"
```

Comment:
IMO, it's better to create a function for each query in db package and use it in services where required.

Reply:
yes sir we have created db functions in latest PR s